### PR TITLE
Tweaks xeno chestbursting

### PR DESCRIPTION
**Pull request by benbot16**
## About The Pull Request

Someone suggested this to me at some point a while ago, but I never got around to coding it in (so sorry to whoever that was).
This PR makes it so that chestbursters won't gib their hosts. Instead, it'll just kill them and make all the organs in their chest spill out.
I also tweaked the logic on alien embryo growths to be a bit more consistent. Now, instead of checking prob(3) every time the egg processes, it just sets a timer from 1-1.5 minutes for each stage and progresses when that timer's up. If the embryo's removed, the stage timer resets. 
There's probably a better way to manage the stage times, but I'm not entirely sure if there's something else I should use. 

## Why It's Good For The Game

Because full gibbing + brain deletion from a worm coming out of you never really made sense to me. That, and getting fully round removed kinda sucks.

The change for growth should also hopefully make embryo growth more consistent. 

## Testing Photographs and Procedure

It compiles, and there's no runtimes or incredibly broken things during embryo processing/chestbursting.

## Changelog
:cl:
tweak: Xeno chestbursters no longer gib their hosts.
tweak: Tweaks alien embryo growth logic
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
People will not pick xeno if they believe they can still roundreturn. I think the braingib's purpose is that.

**Comment by FriendlyContractor** [comment]
Bad idea. Bursted players are intentionally made unrevivable so they are encouraged to join as xenomorph and leaving them revivable will lead to various weird situations about which i will not go into detail here and most importantly can lead to degenerated situation where xenomorph hive can not grow because there are no players willing to take the role

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm [hunk 1]
```diff
@@ -5,6 +5,7 @@
 	icon = 'icons/mob/alien.dmi'
 	icon_state = "larva0_dead"
 	var/stage = 0
+	var/next_stage_time = 0
```
Comment (Archanial) [inline code/modules/mob/living/carbon/alien/special/alien_embryo.dm]: Use cooldown defines

**Review by Archanial — DISMISSED** [review]

**Comment by williamcll** [comment]
Actually show accurate.

**Comment by PerishedFraud** [comment]
I approve of this change because xenos are 

1 unfun to play against, relying entirely on counter items

2 still unfun to play against even if you have those counter items

3 haha xeno go brr at methspeed while almost invisible

4 did i mention they're bad

5 yes

File: code/modules/mob/living/carbon/alien/special/alien_embryo.dm [hunk 2]
```diff
@@ -50,24 +51,33 @@
 			to_chat(owner, "<span class='danger'>You feel something tearing its way out of your stomach.</span>")
 			owner.adjustToxLoss(10)
 
+/obj/item/organ/body_egg/alien_embryo/on_death()
+	. = ..()
+	if(!owner && next_stage_time) // No growing outside the body
+		COOLDOWN_RESET(src, next_stage_time)
```
Comment (Archanial) [inline code/modules/mob/living/carbon/alien/special/alien_embryo.dm]: stop processing/kill/delete the organ instead

Comment (benbot16) [reply code/modules/mob/living/carbon/alien/special/alien_embryo.dm]: This should do it.

**Review by Archanial — APPROVED** [review]

**Comment by PowerfulBacon** [comment]
What does this actually do to the corpse though?

**Comment by benbot16** [comment]
> What does this actually do to the corpse though?

It dismembers the chest and deals a lot of damage to it to kill the host.

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**